src/Client.php

```diff
@@ -489,6 +469,158 @@ public function updateUser(User $user) {
+                $errorsAsString .= $message;
+                $errorsAsString .= ', ';
+            }
+            throw new SmarterUException($errorsAsString);
+        }
+
+        $groups = [];
+        foreach ($bodyAsXml->Info->UserGroups->children() as $group) {
+            $currentGroup = [];
+            $permissions = [];
+            foreach ($group->Permissions->children() as $permission) {
+                $permissions[] = (string) $permission;
+            }
+            $currentGroup['Name'] = $group->Name;
+            $currentGroup['Identifier'] = $group->Identifier;
+            $currentGroup['IsHomeGroup'] = $group->IsHomeGroup;
+            $currentGroup['Permissions'] = $permissions;
+            $groups[] = $currentGroup;
+        }
+
+        $result = [
+            'Response' => $groups,
+            'Errors' => $errorMessages
+        ];
+
+        return $result;
+    }
+
+    /**
+     * Make a CreateGroup query to the SmarterU API.
+     *
+     * @param Group $group The Group to create
+     * @return array An array of [$result, $errors] where $result is an array
+     *      of any information returned by the SmarterU API and $errors is an
+     *      array of any error messages returned by the SmarterU API.
+     * @throws MissingValueException If the Account API Key and/or the User
+     *      API Key are unset.
+     * @throws ClientException If the HTTP response includes a status code
+     *      indicating that an HTTP error has prevented the request from
+     *      being made.
+     * @throws SmarterUException If the response from the SmarterU API
+     *      reports a fatal error that prevents the request from executing.
+     */
+    public function createGroup(Group $group): array  {
+        $xml = $group->toXml(
+            $this->getAccountApi(),
+            $this->getUserApi(),
+            'createGroup'
+        );
+
+        if (empty($this->getHttpClient())) {
+            $this->setHttpClient(new HttpClient([
+                'base_uri' => self::POST_URL
+            ]));
+        }
+
+        $response = $this
+            ->getHttpClient()
+            ->request('POST', self::POST_URL, ['package' => $xml]);
+
+        $body = (string) $response->getBody();
+        $bodyAsXml = simplexml_load_string($body);
+
+        $result = (string) $bodyAsXml->Result;
+
+        $errorMessages = [];
+        $errors = $bodyAsXml->Errors;
+        if ($errors->count() !== 0) {
+            $errorMessages = $this->readErrors($errors);
+        }
+
+        if (strcmp($result, 'Failed') == 0) {
+            $errorsAsString = 'SmarterU rejected the request due to the following error(s): ';
+            foreach ($errorMessages as $id => $message) {
+                $errorsAsString .= $id;
+                $errorsAsString .= ": ";
+                $errorsAsString .= $message;
+                $errorsAsString .= ", ";
+            }
+            throw new SmarterUException($errorsAsString);
```

Comment:
I do think we should just add `$errorMessages` as 'details' here. Of course I've got my head in some other language so you probably thought I was crazy saying that on PR#5. Tell you what I'll do a PR so that the suggestion below will work.

```suggestion
            throw new SmarterUException($errorsAsString, $errorMessages);
```